### PR TITLE
Use geary systems instead of manual checks

### PR DESCRIPTION
**Pull request by DevScyu**
A better way for applying effects to the player

(Will need players to re-set their spawn at their campfire to get the effect applied again)

### CODE REVIEW

**Review by Norazan — COMMENTED** [review]
Don't know much about the ECS TickingSystem but might it be beneficial to remove the BonfireEffectArea component from players if they unset their respawn or the respective bonfire is broken? This might reduce the number of ticks processed by the BonfireEffectSystem. Idk if there is a substantial performance improvement tho.

**Review by Norazan — APPROVED** [review]
nvm I was blind

File: src/main/java/com/mineinabyss/bonfire/ecs/systems/BonfireEffectSystem.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    override fun QueryResult.tick() {
+        // Check if still near a bonfire
+        player.location.getNearbyLivingEntities(BonfireConfig.data.effectRadius).firstOrNull {
+            it is ArmorStand && it.isBonfireModel() && it.uniqueId == effect.uuid
+        }?.let {
+            player.location.world.spawnParticle(
+                listOf(Particle.SOUL, Particle.SOUL_FIRE_FLAME).random(),
+                player.location,
+                1,
```
Comment (0ffz) [inline src/main/java/com/mineinabyss/bonfire/ecs/systems/BonfireEffectSystem.kt:25]: I generally like explicitly adding the parameter names when they go on newlines (you just have so much extra space).

**Review by 0ffz — APPROVED** [review]
Just one small comment, otheriwse looks good

**Pull request merged**